packages/xrpl/src/sugar/submit.ts

```diff
@@ -119,29 +120,42 @@ async function submitRequest(
 async function waitForFinalTransactionOutcome(
   client: Client,
   txHash: string,
+  lastLedger: number,
 ): Promise<TxResponse> {
   await sleep(LEDGER_CLOSE_TIME)
 
-  const txResponse = await client.request({
-    command: 'tx',
-    transaction: txHash,
-  })
+  const latestLedger = await client.getLedgerIndex()
+
+  const txResponse = await client
+    .request({
+      command: 'tx',
+      transaction: txHash,
+    })
+    .catch(async (error) => {
+      // this type of error does not have its interface, type-assert and extract the value we need.
+      // eslint-disable-next-line @typescript-eslint/consistent-type-assertions,@typescript-eslint/no-unsafe-member-access -- L131
+      const message = error.data.error as string
+      if (message === 'txnNotFound') {
+        if (lastLedger < (await client.getLedgerIndex())) {
```

Comment:
This should use `latestLedger`

Reply:
latest is the the value that is being computed on the RHS.

Reply:
so if last<latest, which means lastLedgerSeq has passed, we throw an error.

Reply:
Yes, why is it being recalculated instead of just using the existing calculated value?

Reply:
got rid of it